rust/tool/src/logging.rs

```diff
@@ -0,0 +1,42 @@
+use log::{LevelFilter, Metadata, Record};
+
+pub struct CliLogger(LevelFilter);
+
+/// A simple logger implementation.
+///
+/// Does not display any extra information besides the log messages themselves.
+///
+/// Only displays log messages from this crate itself.
+impl CliLogger {
+    pub fn init(level: LevelFilter) {
+        log::set_boxed_logger(Box::new(Self(level))).expect("Failed to setup logger.");
+        log::set_max_level(level);
+    }
+}
+
+impl log::Log for CliLogger {
+    fn enabled(&self, metadata: &Metadata) -> bool {
+        metadata.level() <= self.0
+    }
+
+    fn log(&self, record: &Record) {
+        if !self.enabled(record.metadata()) {
+            return;
+        };
+        // Do not write the log if it does not come from this crate.
+        if !lzbt_record(record) {
+            return;
+        }
+        println!("{}", record.args());
```

Comment:
Typically logging goes to stderr to avoid buffering log messages. But at that point, you just re-implement [stderrlog](https://docs.rs/stderrlog/0.5.4/stderrlog/). Why not use that? It also gives you the typical Unix behavior of `-v`, `-vv`, `-vvv`.

Reply:
Never heard of the stderrlog crate before but it looks good. I'm not too fond of the multiple -vvv flags, since I do not believe that a normal CLI user would be interested in anything besides normal output and debug output. However, since Lanzaboote is not really a normal CLI, as it is only called by a module, we might as well implement all log levels.